pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -544,3 +720,18 @@ var _ = Describe("applySynchronizedConditionWithPatch", func() {
 	})
 
 })
+
+// awsProviderSpecFromMachine unmarshals a the raw extension into an AWSMachineProviderSpec type.
+func awsProviderSpecFromMachine(mapiMachine *machinev1beta1.Machine) (machinev1beta1.AWSMachineProviderConfig, error) {
+	rawExtension := mapiMachine.Spec.ProviderSpec.Value
+	if rawExtension == nil {
+		return machinev1beta1.AWSMachineProviderConfig{}, nil
+	}
+
+	spec := machinev1beta1.AWSMachineProviderConfig{}
+	if err := yaml.Unmarshal(rawExtension.Raw, &spec); err != nil {
+		return machinev1beta1.AWSMachineProviderConfig{}, fmt.Errorf("error unmarshalling providerSpec: %w", err)
+	}
+
+	return spec, nil
+}
```

Comment:
Can we wrap/use `mapi2capi.AWSProviderSpecFromRawExtension` then?

Reply:
I'm not actually sure here. 

I'm using it as follows:  

```go
It("should update the MAPI provider spec", func() {
  Eventually(k.Object(mapiMachine), timeout).Should(
    WithTransform(awsProviderSpecFromMachine,
      HaveField("InstanceType", Equal("m7i.4xlarge")),
      ))
})
```

I'm not sure how to pass  `mapiMachine.Spec.ProviderSpec.Value` to `withTransform` in this instance, keeping the `Eventually` over the `k.Object`.

Reply:
Ohh - I see. Sure, makes sense.